macOS: Various keyboard input fixes following merge of NSEvent driver

sukibaby opened the pull request:
Now that I've been able to test on macOS a bit more, I found a few more things that need fixing with the NSEvent keyboard input driver. They are all addressed here.

## 1. Screenshot key was not responsive
  -  `/` key functionality is restored as expected.
## 2. Improve modified key event handling
  -  Relying on `keyCode` may not distinguish multiples of the same key (left ctrl vs right ctrl for example), so the logic is made more robust.
  -  Added a flag to manually keep track of previous modifier.
## 3. Fix overlap with Mission Control shortcuts
  - In Edit Mode, Ctrl+Up and Ctrl+Down to change the zoom level were unusable because this is a systemwide hotkey for Mission Control. For macOS only, I've added `ifdef`s into ScreenEdit to instead use Alt+Up and Alt+Down for the zoom level commands. This seems to be reflected automatically in the F1 overlay:
  - ![2025-11-11_012320](https://github.com/user-attachments/assets/38ad60ae-ea31-4ccf-9847-1bbdfb4cf938)

sukibaby committed:
Restore screenshot key to NSEvent input driver

sukibaby committed:
macOS: Add Ctrl+Shift requirement to avoid overlap with Mission Control shortcuts

macOS uses the Ctrl+Up and Ctrl+Down shortcuts system wide for Mission Control, so we can't use it for Edit Mode. This changes the macOS binding for those controls to include shift (Alt+Up and Alt+Down).

teejusb commented on src/arch/InputHandler/InputHandler_NSEvent.mm:
```diff
@@ -174,31 +178,64 @@ static float DownOrUp( bool down )
-            ButtonPressed( DeviceInput( DEVICE_KEYBOARD, m_NSKeyCodeMap[keyCode], DownOrUp( flags & NSEventModifierFlagFunction ), now ) );
-            break;
-        }} break;
+        NSEventModifierFlags changedFlags = flags ^ g_PreviousModifierFlags;
```
I think this looks a lot uglier and I don't know if this is even necessary.

sukibaby replied:
Reverting this commit

teejusb commented:
> use Alt+Up and Alt+Down for the zoom level commands. This seems to be reflected automatically in the F1 overlay:

*technically* on mac, `alt` is known as the `option` key. I wonder if there's a way to reflect that in the help text just on mac. It's not the worst thing, but macs don't visibly have an `alt` key on the keyboard.

sukibaby commented:
> > use Alt+Up and Alt+Down for the zoom level commands. This seems to be reflected automatically in the F1 overlay:
> 
> _technically_ on mac, `alt` is known as the `option` key. I wonder if there's a way to reflect that in the help text just on mac. It's not the worst thing, but macs don't visibly have an `alt` key on the keyboard.

Done, nice and easy.

<img width="1392" height="860" alt="Screenshot 2025-11-26 at 11 15 27 PM" src="https://github.com/user-attachments/assets/560f177e-65f2-40e6-abd2-be10db77cf47" />

sukibaby committed:
macOS: Update help button labels

On macOS, replaces 'Ctrl' with 'Cmd' and 'Alt' with 'Option' in Edit Mode's help menu (F1 key) to match platform conventions.
